analytics_data_api/management/commands/generate_data.py

```diff
@@ -0,0 +1,339 @@
+# pylint: disable=line-too-long,invalid-name
+"""
+Abstract base management command meant to be called by other commands that generate fake data
+"""
+
+import abc
+import datetime
+import logging
+import math
+import random
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+from tqdm import tqdm
+
+from analytics_data_api.constants import engagement_events
+from analytics_data_api.v0 import models
+from analyticsdataserver.clients import CourseBlocksApiClient
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+# http://stackoverflow.com/a/3590105
+def constrained_sum_sample_pos(num_values, total):
+    """Return a randomly chosen list of n positive integers summing to total.
+    Each such list is equally likely to occur."""
+
+    dividers = sorted(random.sample(range(1, total), num_values - 1))
+    return [a - b for a, b in zip(dividers + [total], [0] + dividers)]
+
+
+def get_count(start):
```

Comment:
this name is not that helpful but I tried a bit to think of a better one and can't

at least change start to base or min?